CONTRIBUTING.md

```diff
@@ -179,3 +179,57 @@ Do your best to follow the clean code guidelines.
 - Don’t return an error code, throw an exception instead.
 
 Ref: Book: Clean Code by Robert C. Martin (Uncle Bob)
+
+## Compilation and Installation
+
+This guide is helpful to compile and install Nmstate from source, for installing stable release or other installation methods, pleaser refer to [Nmstate installation guide](https://nmstate.io/user/install.md)
+
+### Dependencies
+
+Nmstate requires NetworkManager 1.26 or later version installed and started.
+
+In order to support specific capabilities, additional packages are required:
+
+OpenVSwitch support through the NM provider requires NetworkManager-ovs.
+
+Extended ovsdb support requires openvswitch-2.11 and python3-openvswitch2.11 or greater.
+
+Team interface support through the NM provider requires NetworkManager-team
+
+### Clone the source
+```
+git clone https://github.com/nmstate/nmstate.git
+cd nmstate
+```
+
+#### Install build tool
+```
+sudo dnf install cargo # Fedora
+sudo yum install cargo # RHEL
+cargo --version
```

Comment:
RHEL uses dnf now, so no need to have 2 different lines for them.
Also, there are other distributions like Ubuntu/Debian that the developer might be using. Better to say something more generic like "install cargo from your distribution's repositories".
And finally, for development is normally prefered to install cargo using "rustup" (search the official documentation to see how). So better to indicate installing with rustup, and distribution's repositories as an alternative.